### PR TITLE
#260: Build API improvements

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/issues/260 |
| Dependencies | https://github.com/ripe-tech/ripe-core/pull/4444 |
| Decisions | Support for new ripe-core endpoints for installing, uninstalling and updating builds. |
| Animated GIF | -- |

### CODE REVIEW

**Comment by gcandal** [comment]
@FlaviooLima  @BeeMargarida this is a command and not a listener, so rename it accordingly (`onBuildUpdate` -> `updateBuild`, etc).

File: src/js/api/brand.js [hunk 1]
```diff
@@ -265,27 +265,6 @@ ripe.Ripe.prototype.onInitials = function(options, callback) {
...
- * a server side implementation of the 3DB's business logic.
- *
- * @param {Object} options An object with options, such as:
- *  - 'group' - The name of the group that is going to be changed
- *  - 'value' - The initials value to be changed
- *  - 'engraving' - The engraving value to be changed
- * @param {Function} callback Function with the result of the request.
- * @returns {Promise} Resulting information for the callback execution.
- */
-ripe.Ripe.prototype.onInitialsP = function(options, callback) {
```
Comment (gcandal) [inline src/js/api/brand.js:281]: Oops?

File: src/js/api/build.js [hunk 2]
```diff
@@ -126,6 +126,145 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
...
+ * a server side implementation of the 3DB's business logic.
+ *
+ * @param {Object} options An object with options, such as:
+ *  - 'group' - The name of the group that is going to be changed
+ *  - 'value' - The initials value to be changed
+ *  - 'engraving' - The engraving value to be changed
+ * @param {Function} callback Function with the result of the request.
+ * @returns {Promise} Resulting information for the callback execution.
+ */
+ripe.Ripe.prototype.onInitialsP = function(options, callback) {
```
Comment (gcandal) [inline src/js/api/build.js:142]: This move was a mistake :p

File: src/js/api/build.js [hunk 3]
```diff
@@ -126,6 +126,145 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
...
+ *
+ * @param {String} name The name of the brand of the build.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.updateBuild = function(name, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}builds/${name}/update`;
+
```
Comment (gcandal) [inline src/js/api/build.js:161]: No newline here, for coherence with the surrounding code

File: src/js/api/build.js (hunk 2 above)
Comment (BeeMargarida) [reply src/js/api/build.js:142]: oops, my bad! Will fix it

File: src/js/api/build.js [hunk 4]
```diff
@@ -126,6 +126,145 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
...
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.uninstallBuild = function(name, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const version = options.version === undefined ? "latest" : options.version;
+    const url = `${this.url}builds/${name}/uninstall`;
+    const params = {};
+    if (version !== undefined && version !== null) {
+        params.version = version;
```
Comment (gcandal) [inline src/js/api/build.js:200]: Document this (check `onInitialsP`, which also documents options).

File: src/js/api/build.js [hunk 5]
```diff
@@ -126,6 +126,145 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
...
+ * Uninstalls a build by brand name and version.
+ *
+ * @param {String} name The name of the brand of the build.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.uninstallBuild = function(name, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const version = options.version === undefined ? "latest" : options.version;
```
Comment (gcandal) [inline src/js/api/build.js:196]: Actually, unlike install, uninstall shouldn't default to `latest` (check https://github.com/ripe-tech/ripe-core/blob/master/src/ripe_core/controllers/api/build.py#L65).

File: src/js/api/build.js [hunk 6]
```diff
@@ -126,6 +126,145 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
...
+ * Installs a build by brand name and version.
+ *
+ * @param {String} name The name of the brand of the build.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.installBuild = function(name, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const version = options.version === undefined ? "latest" : options.version;
```
Comment (gcandal) [inline src/js/api/build.js:237]: Document this.

File: src/js/api/build.js [hunk 7]
```diff
@@ -126,6 +126,133 @@ ripe.Ripe.prototype.getBuildArtifactP = function(name, version, options) {
     });
 };
 
+/**
+ * Updates a build by brand name and version.
+ *
+ * @param {String} name The name of the brand of the build.
+ * @param {Object} options An object with options, such as:
+ *  - 'version' - The version of the build
```
Comment (gcandal) [inline src/js/api/build.js:134]: No longer true, right?

**Pull request merged**